src/wobjectcpp.h

```diff
@@ -0,0 +1,161 @@
+struct MetaMethodInfoBuilder {
+    StringView name;
+    F func;
+    ParamTypes paramTypes{};
+    ParamNames paramNames{};
+    static constexpr int flags = Flags;
+    using IntegralConstant = IC;
+
+#if __cplusplus > 201700L
+    template<class... Args, class = std::enable_if_t<(std::is_same_v<std::decay_t<Args>, StringView> && ...)>>
+#else
+    template<class... Args, class = std::enable_if_t<w_internal::all<std::is_same<std::decay_t<Args>, StringView>::value...>()>>
+#endif
+    constexpr auto setParamTypes(Args... paramTypes) const
+        -> MetaMethodInfoBuilder<F, Flags, IC, w_internal::StringViewArray<sizeof... (Args)>, ParamNames> {
+        return {name, func, {paramTypes...}, paramNames};
+    }
+#if __cplusplus > 201700L
+    template<class... Args, class = std::enable_if_t<(std::is_same_v<std::decay_t<Args>, StringView> && ...)>>
+#else
+    template<class... Args, class = std::enable_if_t<w_internal::all<std::is_same<std::decay_t<Args>, StringView>::value...>()>>
+#endif
+    constexpr auto setParamNames(Args... paramNames) const
+        -> MetaMethodInfoBuilder<F, Flags, IC, ParamTypes, w_internal::StringViewArray<sizeof... (Args)>> {
+        return {name, func, paramTypes, {paramNames...}};
+    }
+    template<int... fs>
+    constexpr auto addFlags(w_internal::W_MethodFlags<fs>...) const
+#if __cplusplus > 201700L
+        -> MetaMethodInfoBuilder<F, (fs | ... | Flags), IC, ParamTypes, ParamNames> {
+#else
+        -> MetaMethodInfoBuilder<F, w_internal::summed<fs ..., Flags>, IC, ParamTypes, ParamNames> {
+#endif
+            return {name, func, paramTypes, paramNames};
+    }
+    template<class IC2>
+    constexpr auto setIntegralConstant() const -> MetaMethodInfoBuilder<F, Flags, IC2, ParamTypes, ParamNames> {
+        return {name, func, paramTypes, paramNames};
+    }
+
+    constexpr auto build() const -> w_internal::MetaMethodInfo<F, Flags, IC, ParamTypes, ParamNames> {
+        return {func, name, paramTypes, paramNames};
+    }
+};
+
+} // namespace w_internal
+
+namespace w_cpp {
+
+/// Very simple constexpr version of std::string_view
+///
+/// example usage:
+///     constexpr char text[6] = "Hello"; // if you have a string literal use `viewLiteral` below.
+///     constexpr auto view = w_cpp::StringView{&text[0], &text[6]};
+///
+/// \note the end pointer has to point behind the \0
+using w_internal::StringView;
+
+/// Generate a constexpr StringView from a C++ string literal.
+///
+/// example usage:
+///     constexpr auto view & w_cpp::StringView{"Hello"};
+using w_internal::viewLiteral;
+
+/// create a compile time property description for registraton usage
+///
+/// \arg T the type of the property
+/// \arg name the compile time StringView for the property name
+/// \arg type the compile time StringView for the type of the property
+///
+/// Following methods may be called in chain:

```

Comment:
maybe mention it is using the "Builder pattern"

Reply:
Its not really a builder pattern, as no build function has to be called. Each step is valid.
If you want I can implement a real builder for properties, just for this new API.
For the first implementation it seemed a bit too much work.

@ogoffart What do you think?

Reply:
I think that's fine as it is.